rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbIT.java

```diff
@@ -0,0 +1,497 @@
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Second vote from a different authorizer
+        voteFeePerKb(fifthFeePerKbVote, FeePerKbVoteCaller.THIRD_AUTHORIZED.getRskAddress());
+        currentFeePerKb = fifthFeePerKbVote;
+
+        // Get fee per kb, should return the newly voted value
+        assertFeePerKbValue(fifthFeePerKbVote);
+    }
+
+    @Test
+    @Order(6)
+    void voteFeePerKbChange_whenAuthorizerOneVotesDifferentValueAndAuthorizerTwoAndThreeVoteSameValue_shouldSetValueVotedByTwoAndThree() {
+        Coin sixthFeePerKbVote = Coin.valueOf(30_000L);
+
+        // First vote for a new value
+        voteFeePerKb(differentFeePerKbVote, FeePerKbVoteCaller.FIRST_AUTHORIZED.getRskAddress());
+        // Get fee per kb, should return the fifthFeePerKbVote voted value because the voting has only 1 vote
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Second vote for a different value, different authorizer
+        voteFeePerKb(sixthFeePerKbVote, FeePerKbVoteCaller.SECOND_AUTHORIZED.getRskAddress());
+        // Get fee per kb, should return the second round voting value because the third value has only 2 different votes
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Third vote from authorizer 3, same value as the second one voted
+        voteFeePerKb(sixthFeePerKbVote, FeePerKbVoteCaller.THIRD_AUTHORIZED.getRskAddress());
+        currentFeePerKb = sixthFeePerKbVote;
+
+        // Get fee per kb, should return sixthFeePerKbVote value
+        assertFeePerKbValue(sixthFeePerKbVote);
+    }
+
+    @Test
+    @Order(7)
+    void voteFeePerKbChange_whenAuthorizerOneAndThreeVoteSameValueAndInBetweenVotesUnauthorizedCallerSameValueThanOne_shouldSetValueWithLastVoteEmittedByThree() {
+        Coin seventhFeePerKbVote = Coin.valueOf(40_000L);
+
+        // First vote for a new value
+        voteFeePerKb(seventhFeePerKbVote, FeePerKbVoteCaller.FIRST_AUTHORIZED.getRskAddress());
+        // Get fee per kb, should return the second voted value because the third value has only 1 vote
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Second vote for an unauthorized caller
+        voteFeePerKb(seventhFeePerKbVote, FeePerKbVoteCaller.UNAUTHORIZED.getRskAddress());
+        //It still has the previous voting round value
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Third vote from authorizer 3, same value as the first one voted from authorizer 1
+        voteFeePerKb(seventhFeePerKbVote, FeePerKbVoteCaller.THIRD_AUTHORIZED.getRskAddress());
+        currentFeePerKb = seventhFeePerKbVote;
+
+        // Get fee per kb, should return the third vote value
+        assertFeePerKbValue(seventhFeePerKbVote);
+    }
+
+    @Test
+    @Order(8)
+    void voteFeePerKbChange_whenAuthorizerOneAndThreeVoteAfterAuthorizerOneVoteAuthorizerTwoDifferentValueAfterThatUnauthorizedCallerVotesSameThanAuthorizerTwoAndLasVoteIsEmittedByAuthorizerThree_shouldSetValueVotedByOneAndThree() {
```

Comment:
kind of nit, but I find this test name really hard to follow

Reply:
Please let me know if this test name is more readable and easier to get:

`voteFeePerKbChange_whenAuthorizerOneAndThreeVoteSameValueThenInBetweenAuthorizerTwoAndUnauthorizedCallerVoteSame_shouldSetValueVotedByOneAndThree`